west.yml

```diff
@@ -143,7 +143,8 @@ manifest:
       groups:
         - hal
     - name: hal_xtensa
-      revision: 0e577021bb66e644afd067cd9f7c71ab11b62b3d
+      url: https://github.com/aborisovich/hal_xtensa.git
```

Comment:
@nashif can you elaborate? On one hand `pull/15/head` is readable. On the other hand the SHA1 is deterministic. I think both can be achieved with:
```
 - name: hal_xtensa
      # https://github.com/zephyrproject-rtos/hal_xtensa/pull/15
      revision: 0e577021bb66e644afd067cd9f7c71ab11b62b3d
      url: https://github.com/aborisovich/hal_xtensa.git
```

Reply:
Manifest changes is always a two-part process. First, set it to the manifest PR so CI can run through it. Once CI passes, the manifest PR can be merged. Then the manifest needs to be updated to point to the SHA. And you let CI run through it again (this time with the actual SHA).

Reply:
I understand it's made of two steps, I'm wondering what's wrong with using (volatile) SHA1s for the first step too.

Reply:
Since the PR has not been merged, normal git fetch/pull won't download that commit, and thus git checkout will fail later. This is due to that SHA not in the tree of the fetched/pulled branch (as west simply does fetch only).

Reply:
https://docs.zephyrproject.org/latest/develop/modules.html?highlight=module#process-for-submitting-a-new-module

Reply:
First, sorry I misread the diff (the diff on the main, "conversation" page is incomplete). I thought a `revision: SHA1` was being submitted.

> Since the PR has not been merged, normal git fetch/pull won't download that commit, and thus git checkout will fail later.

With Github it works as long as the SHA1 is 40 digits long and in the fork (github shared storage). It's magic, demo:

```diff
--- a/west.yml
+++ b/west.yml
@@ -143,7 +143,8 @@ manifest:
       groups:
         - hal
     - name: hal_xtensa
-      revision: 0e577021bb66e644afd067cd9f7c71ab11b62b3d
+      # https://github.com/aborisovich/hal_xtensa/commits/add-soc-overlay-for-intel_ace15_mtpm
+      revision: cac06cfebfd36a6f5cf05ca13774c36380ff1dce
       path: modules/hal/xtensa
       groups:
         - hal
```

```shell
west update -n hal_xtensa
=== updating hal_xtensa (modules/hal/xtensa):
--- hal_xtensa: fetching, need revision cac06cfebfd36a6f5cf05ca13774c36380ff1dce
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (10/10), done.
remote: Total 14 (delta 2), reused 14 (delta 2), pack-reused 0
Unpacking objects: 100% (14/14), 22.67 KiB | 7.56 MiB/s, done.
From https://github.com/zephyrproject-rtos/hal_xtensa
 * branch                      cac06cfebfd36a6f5cf05ca13774c36380ff1dce -> FETCH_HEAD
M	zephyr/soc/intel_apl_adsp/xtensa/config/core-isa.h
Previous HEAD position was 0e577021bb66 zephyr: add SoC overlay for NXP i.MX8MP
HEAD is now at cac06cfebfd3 zephyr: add SoC overlay for intel_ace15_mtpm
```

Check https://github.com/thesofproject/rimage/pull/71 for more details.

> This clearly links to the PR and shows that a PR needs to be reviewed and merged first.

Thx.